### PR TITLE
Added project and repository definitions to Glossary

### PR DESCRIPTION
**Pull request by claredillon**
Following an actions from an ISPO WG meeting in May 2024, I have added definitions for project and repository to the Glossary. I re-ordered the existing entries to make them alphabetical.

### CODE REVIEW

**Comment by claredillon** [comment]
@dellagustin @jeffabailey This is following up on our ISPO WG discussion - can you check to see if the definitions match your understanding please?

File: meta/glossary.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ This glossary defines essential terms that are commonly used when writing new pa
 
 Through this the glossary helps members of the Patterns Working Group to use consistent language when writing and reviewing patterns.
 
-- **organization** - An umbrella for multiple legal entities. (synonyms: group, enterprise) (e.g. Lufthansa)
 - **legal entity** - An entity that has its own legal rights and obligations (synonyms: company, subsidiary) (e.g. Lufthansa Systems GmbH, Lufthansa Industry Solutions TS GmbH, ...)
+- **organization** - An umbrella for multiple legal entities. (synonyms: group, enterprise) (e.g. Lufthansa)
+- **project** - Projects serve as a planning and tracking tool that can integrate with the issues and pull requests within one or more repositories. In [GitHub documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects), projects are described as an adaptable spreadsheet, task-board, and road map that integrates with your issues and pull requests to help you plan and track your work effectively.  A single project can include items (e.g. issues and pull requests) from multiple repositories.
```
Comment (spier) [inline meta/glossary.md]:
```suggestion
- **project** - Projects serve as a planning and tracking tool that can integrate with the issues and pull requests within one or more repositories. In [GitHub documentation](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects), projects are described as an adaptable spreadsheet, task-board, and road map that integrates with your issues and pull requests to help you plan and track your work effectively.  A single project can include items (e.g. issues and pull requests) from multiple repositories.
```

**Review by jeffabailey — APPROVED** [review]
Looks great!

**Review by jeffabailey — APPROVED** [review]
Looks great!

**Comment by spier** [comment]
@claredillon @jeffabailey thank you for your work on this!

I took this PR as an opportunity to finally get the glossary added to the book:
https://patterns.innersourcecommons.org/appendix/glossary

While it is still small, it is a start :)

**Comment by dellagustin-sap** [comment]
Thank you @claredillon , a point for future improvement is to clarify how the terms organization, repository and project are used across different source code management (SCM) systems (e.g., GitHub, GitLab, Gitea, Forgejo, BitBucket, etc...).
There is also some ambiguity on the term organization that we need to sort out (organization as currently described vs organization as used in SCMs.

**Comment by claredillon** [comment]
@dellagustin-sap - thanks for the feedback - would love to explore that ambiguity more... I wonder if we could get others more familiar with the other SCMs to help out... I am only familiar with GitHub.

**Comment by claredillon** [comment]
Thanks @spier and @jeffabailey for the reviews and merging

**Pull request merged**